### ISSUE TITLE
scripts.changelog does not actually executed in --dry-run mode

### ISSUE THREAD

**Issue by 1natsu172** [webpro/release-it#472]
Thanks for the best release tool.

Version: `release-it@10.1.0`

part of `.release-it.json`

```json
  "scripts": {
    "changelog": "conventional-changelog -p angular | tail -n +3",
    "beforeStage": "conventional-changelog -p angular -i CHANGELOG.md -s",
    "afterRelease": "echo -e \"Successfully released ${name} v${version} to ${repo.repository}.\n\nNext: Let's write release notes on GitHub :)\"; open https://github.com/1natsu172/Outside-YouTube-Player-Bar/releases"
  },
```

---

I want to see the content of CHANGELOG before release, so I run with `--dry-run`, but it is displayed  `Empty changelog`.(Actually not empty. After production release, CHANGELOG was output properly…)

![](https://i.gyazo.com/e2090e54cb77c86894fb12b314d65e82.png)

I can understand that the `beforeStage` and `afterRelease` commands will not be executed because it can not handle on "release-it" side… 🤔 

Is it intentional that `scripts.changelog` is not executed? (or is it wrong to check CHANGELOG by dry-run?)

**Comment by webpro** [comment]
Hi @1natsu172! It intentionally does not execute any custom user scripts from `scripts.*` as they *may* write things, although `scripts.changelog` *should* only read.

So yeah, I agree a changelog should be generated in a dry run as well. You up for a PR with a fix? :-)

**Comment by 1natsu172** [comment]
Thanks for fast reply!

First of all I wanted to confirm to you whether this behavior was intentional.

> You up for a PR with a fix? :-)

I think will submit fix PR tomorrow, if can do it well 😃

**Comment by webpro** [comment]
Released in [v10.2.0](https://github.com/webpro/release-it/releases/tag/10.2.0)
